### PR TITLE
WT-6477 format.sh ignores failures if run directory never created

### PR DESCRIPTION
**Pull request by keithbostic**

### CODE REVIEW

**Comment by lukech** [comment]
@keithbostic It looks you are trying to redesign the handling of RUNDIR-directory-not-created case in `format.sh`. I believe the initial purpose of WT-6477 is to address the silent Evergreen failure (with a core dump generation), and the reason the task did not fail was `setsid` made the return value be 0 though the command (`./t`) it prefixes failed, as described in the jira [comment](https://jira.mongodb.org/secure/EditComment!default.jspa?id=1389645&commentId=3223162). To be able to detect the failures we'll need to check into the log file. 

As long as we keep using `setsid`, there's a risk we could miss detecting failures (initially RUNDIR-directory-not-created case, recently rebalance failure with timestamps, could be more).

**Comment by keithbostic** [comment]
@lukech 
> It looks you are trying to redesign the handling of RUNDIR-directory-not-created case in format.sh. I believe the initial purpose of WT-6477 is to address the silent Evergreen failure (with a core dump generation), and the reason the task did not fail was setsid made the return value be 0 though the command (./t) it prefixes failed, as described in the jira comment. To be able to detect the failures we'll need to check into the log file.

I think fixing the "RUNDIR not created case" is what we want, and this change includes checking the log file even if RUNDIR is never created (we previously skipped log files when there was no RUNDIR).

Regardless, it sounds like you see a better solution, and I'm happy to have you discard this branch and move forward with a different approach.

**Comment by lukech** [comment]
@keithbostic I re-thought the problem and realized it's another case of "RUNDIR not created" (besides WT-6172), and your branch is trying to address this category of problems with a common solution, which is reasonable to me. 

I did investigate `setsid` alternatives and could not find any, so I'm happy to go with your approach in this PR.

File: test/format/format.sh [hunk 1]
```diff
@@ -449,7 +456,7 @@ format()
 
 	# Disassociate the command from the shell script so we can exit and let the command
 	# continue to run.
-	# Run format in its own session so child processes are in their own process gorups
+	# Run format in its own session so child processes are in their own process groups
 	# and we can individually terminate (and clean up) running jobs and their children.
 	nohup setsid $cmd > $log 2>&1 &
```
Comment (lukech) [inline test/format/format.sh:462]: Are lines 463-471 (checking log file for signature 'setsid: failed to execute' and marking failure) made redundant with this solution?

Comment (keithbostic) [reply test/format/format.sh:462]: I think it's still needed.

I think there's a window where we don't handle things correctly, in-between calling `setsid` and seeing`running process` in the log file. I moved the code in `format` that writes that line to the log file to be almost the first thing `format` does, but if for some reason the exec just fails, we wouldn't handle it correctly. And, this line tries to get that correct by looking for an exec failure.

There's going to be a window no matter what we do, but I think we've tightened it down pretty hard.

Comment (lukech) [reply test/format/format.sh:462]: Thanks for the explanation.

**Pull request merged**